masonry/src/widgets/indexed_stack.rs

```diff
@@ -0,0 +1,382 @@
+        let widget = IndexedStack::new();
+        let window_size = Size::new(200.0, 200.0);
+        let mut harness =
+            TestHarness::create_with_size(default_property_set(), widget, window_size);
+
+        assert_render_snapshot!(harness, "indexed_stack_empty");
+
+        harness.edit_root_widget(|mut stack| {
+            let mut stack = stack.downcast::<IndexedStack>();
+            IndexedStack::add_child(&mut stack, button::Button::new("A"));
+        });
+        assert_render_snapshot!(harness, "indexed_stack_single");
+
+        harness.edit_root_widget(|mut stack| {
+            let mut stack = stack.downcast::<IndexedStack>();
+            IndexedStack::add_child(&mut stack, button::Button::new("B"));
+            IndexedStack::add_child(&mut stack, button::Button::new("C"));
+            IndexedStack::add_child(&mut stack, button::Button::new("D"));
+        });
+        assert_render_snapshot!(harness, "indexed_stack_many_0");
+
+        harness.edit_root_widget(|mut stack| {
+            let mut stack = stack.downcast::<IndexedStack>();
+            IndexedStack::set_active_child(&mut stack, 2);
+        });
+        assert_render_snapshot!(harness, "indexed_stack_many_2");
+    }
+
+    #[test]
+    fn test_widget_removal_and_modification() {
+        let widget = IndexedStack::new()
+            .with_child(button::Button::new("A"))
+            .with_child(button::Button::new("B"))
+            .with_child(button::Button::new("C"))
+            .with_active_child(1);
+        let window_size = Size::new(200.0, 200.0);
+        let mut harness =
+            TestHarness::create_with_size(default_property_set(), widget, window_size);
+        // Snapshot with the single widget.
+        assert_render_snapshot!(harness, "indexed_stack_initial_builder");
+
+        // Remove the first (inactive) widget
+        harness.edit_root_widget(|mut stack| {
+            let mut stack = stack.downcast::<IndexedStack>();
+            IndexedStack::remove_child(&mut stack, 0);
+        });
+        assert_render_snapshot!(harness, "indexed_stack_initial_builder"); // Should not be changed
+
+        // Now remove the active widget
+        harness.edit_root_widget(|mut stack| {
+            let mut stack = stack.downcast::<IndexedStack>();
+            IndexedStack::remove_child(&mut stack, 0);
+        });
+        assert_render_snapshot!(harness, "indexed_stack_builder_removed_widget");
+
+        // Add another widget at the end
+        harness.edit_root_widget(|mut stack| {
+            let mut stack = stack.downcast::<IndexedStack>();
+            IndexedStack::add_child(&mut stack, button::Button::new("D"));
+        });
+        assert_render_snapshot!(harness, "indexed_stack_builder_removed_widget"); // Should not change
+
+        // Make it active
+        harness.edit_root_widget(|mut stack| {
+            let mut stack = stack.downcast::<IndexedStack>();
+            IndexedStack::set_active_child(&mut stack, 1);
```

Comment:
This is partly what I mean - the new item you've added being index 1 is clear when you follow the logic, but it's not at all the easiest to understand.